dapp/registry/streams.go

```diff
@@ -19,13 +18,12 @@ func (r *Registry) devStreamHandler(str net.Stream) {
 	go func() {
 
 		reader := bufio.NewReader(str)
-		decoder := protoMc.Multicodec(nil).Decoder(reader)
 
 		for {
 
-			// decode app from stream
-			msg := pb.Message{}
-			if err := decoder.Decode(&msg); err != nil {
+			// read DApp from stream
+			jsonDAppStr, err := reader.ReadBytes(0x0A)
```

Comment:
fixed `Additionally, is using 0x0A (New Line) intentional and safe to use in all edge cases?` it's safe since we will receive our data encoded as base64 which doesn't have the `\n` character